source/extensions/filters/common/expr/evaluator.cc

```diff
@@ -55,6 +70,11 @@ BuilderPtr createBuilder(Protobuf::Arena* arena) {
     throw CelException(
         absl::StrCat("failed to register built-in functions: ", register_status.message()));
   }
+
+  if (custom_library) {
```

Comment:
Do we need to account for 'replace_default_library' here? if not, may want to add a comment about why.

Reply:
will add a comment

Reply:
Why do we even have the `replace_default_library` option then?

Reply:
There are functions which are built-in to CEL (https://github.com/google/cel-spec/blob/master/doc/langdef.md#standard-definitions).  I don't think anyone would want to replace those.  

Then are envoy attributes (https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/advanced/attributes) which someone could potentially want to override with custom definitions.

Reply:
This seems good to me! (may be my fault for suggesting the idea of a knob to tweak this in the first place, some months back)

Reply:
Ah, right, that makes sense. Maybe this can be resolved if you share this explanation of the distinction between CEL and Envoy built-ins in the proto documentation for `replace_default_library_in_case_of_overlap`.